bond_ports.go

```diff
@@ -0,0 +1,159 @@
+package packngo
+
+import "fmt"
+
+const NetworkTypeHybridBonded = "hybrid-bonded"
+
+func (d *Device) GetBondNetworkType(portName string) string {
+	for _, p := range d.NetworkPorts {
+		if p.Name == portName {
+			return p.NetworkType
+		}
+	}
+	return ""
+}
+
+func (d *Device) GetEthPortsInBond(name string) []*Port {
+	ports := []*Port{}
+	for _, port := range d.NetworkPorts {
+		if port.Bond != nil && port.Bond.Name == name {
+			ports = append(ports, &port)
+		}
+	}
+	return ports
+}
+
+func BondStateTransitionNecessary(type1, type2 string) bool {
+	if type1 == type2 {
+		return false
+	}
+	if type1 == NetworkTypeHybridBonded && type2 == NetworkTypeL3 {
+		return false
+	}
+	if type2 == NetworkTypeHybridBonded && type1 == NetworkTypeL3 {
+		return false
+	}
+	return true
+}
+
+func (i *DevicePortServiceOp) BondToNetworkType(deviceID, bondPortName, targetType string) (*Device, error) {
+	d, _, err := i.client.Devices.Get(deviceID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	curType := d.GetBondNetworkType(bondPortName)
+
+	if !BondStateTransitionNecessary(curType, targetType) {
+		return nil, fmt.Errorf("Bond doesn't need to be converted from %s to %s", curType, targetType)
```

Comment:
same type transitions should not trigger an error

Reply:
OK!